src/helics/shared_api_library/MessageFederateExport.cpp

```diff
@@ -402,6 +403,7 @@ helics_message_object helicsEndpointGetMessageObject(helics_endpoint endpoint)
     auto message = endObj->endPtr->getMessage();
 
     if (message) {
+        message->messageValidation = messageKeyCode;
```

Comment:
What is the messageValidation trying to detect?

Reply:
the most common case would be if someone passed a federate in what should have been a message object.  So all the objects have a coded sequence in one of the fields to have a first order check against an incorrect argument being passed.  The most likely of which is a different helics object passed incorrectly, since the compiler can't detect that in C like it would in C++, we need a run time check.

Reply:
Okay, so it's not networking related.

Reply:
no, It is using a gap in the structure and the rest of HELICS will ignore it.  Primarily for shared library to detect mistakes at the first level.  All the validation keys in the object, are just there for object checks.